### PR TITLE
Fix super annoying Python 2.6 multiprocessing.Queue stack trace in CI

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

A bug exists in Python 2.6 that sometimes raises an exception during interpreter shutdown. We encounter this frequently in our CI since we run tests on CentOS 6 as the control node, which has Python 2.6.6 with this bug.

This PR adds a very minor sleep only on Python 2.6 which gets around this issue. I did a lot of testing using a standalone script I found that easily duplicated the issue to find the minimum sleep value needed to avoid this issue.

CPython issue: https://bugs.python.org/issue4106
Fix in CPython: https://hg.python.org/cpython/rev/d316315a8781

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

`lib/ansible/executor/task_queue_manager.py`

##### ADDITIONAL INFORMATION

Here is the test script I used in our [CentOS 6 test container](https://quay.io/repository/ansible/centos6-test-container):

```python
#!/usr/bin/env python
#
# This code is from
# https://groups.google.com/g/comp.lang.python/c/QUSUhGMLdv8/m/crEsBKYMxlgJ

# Shell command to test. This needs to be done in a script because it's the
# interpreter shutdown that triggers the race condition.
#
# export i=1; while true; do printf "%-'6d:" "${i}";  python queue-test.py ; let "i+=1" ; done

import sys
import time
from multiprocessing import Process, Queue


def listTest(q):
    print(q.get())


def queueTest():
    q = Queue()
    q.put([1, 2, 3, 4, 5, 6])
    p = Process(target=listTest, args=(q,))
    p.start()
    p.join()


if __name__ == '__main__':
    queueTest()
    # Adding this sleep prevents the traceback
    if (2, 6) == (sys.version_info[0:2]):
        time.sleep(0.0001)
```

I'm not sure about adding tests for this since I haven't found a way to reliably duplicate it running Ansible.

### CODE REVIEW

**Comment by samdoran** [comment]
I am _pretty sure_ this is the correct place to add the `sleep()`. Let me know if I'm way off.

**Review by bcoca — APPROVED** [review]

File: lib/ansible/executor/task_queue_manager.py [hunk 1]
```diff
@@ -328,6 +329,22 @@ def cleanup(self):
...
+        # interpreter shutdown. This is only an issue in our CI testing but we
+        # hit it frequently enough to add a small sleep to avoid the issue.
+        # This can be removed once we have split controller available in CI.
+        #
+        # Further information:
+        #     Issue: https://bugs.python.org/issue4106
+        #     Fix:   https://hg.python.org/cpython/rev/d316315a8781
+        #
+        try:
+            if (2, 6) == (sys.version[0], sys.version[1]):
```
Comment (mattclay) [inline lib/ansible/executor/task_queue_manager.py]: This should be:

```suggestion
            if (2, 6) == (sys.version_info[0:2]):
```

**Review by mattclay — CHANGES_REQUESTED** [review]

**Review by mattclay — APPROVED** [review]
